src/ppc/src/compile.rs

```diff
@@ -226,23 +238,33 @@ impl<'a> ProblemBuilder<'a> {
                         Ok(())
                     });
                 }
-                let status = cmd
+                let output = cmd
                     .stdin(crate::Stdio::piped())
                     .stdout(crate::Stdio::piped())
-                    .status()
-                    .unwrap_or_else(|err| panic!("launching gen_answers cmd failed: {}", err));
-                if !status.success() {
+                    .stderr(crate::Stdio::piped())
+                    .output()
+                    .unwrap_or_else(|err| panic!("launch main solution error: {}", err));
+                if !output.status.success() {
                     eprintln!(
-                        "Error when generating correct answer for test {}: primary solution failed",
+                        "Error when generating correct answer for test {}: main solution failed",
```

Comment:
Should be while generating, not when.
Look at [this](https://forum.wordreference.com/threads/when-doing-vs-while-doing.1568535/).